src/validate-import.ts

```diff
@@ -0,0 +1,26 @@
+import type { Env } from "./env"
+import { normalizeRelativePath } from "./utils"
+import { type Validators, validateId } from "./validate-id"
+
+export function validateImport({
+  id,
+  importValidators,
+  root,
+  importer,
+  env,
+}: {
+  id: string
+  importValidators: Validators
+  root: string
+  importer: string | undefined
+  env: Env
+}): void {
+  validateId({
+    id,
+    env,
+    validators: importValidators,
+    errorMessage: `Import from "${id}"${
+      importer ? ` in "${normalizeRelativePath(root, importer)}"` : ""
+    } is not allowed in the ${env} module graph`,
```

Comment:
Could you ask specific wording for when the `importer` is `undefined` that more clearly explains the failure case for entrypoints?

Reply:
@markdalgleish and I chatted and he made the point that entrypoints (`importer` is undefined) don't make sense to validate here